public/js/main.js

```diff
@@ -0,0 +1,45 @@
+//Animating chat bubbles on conversation cards:
+
+window.onload = function() {
+  //find all convo cards and loop through them
+  let convoCards = document.getElementsByClassName('conversation-card')
+
+  Array.prototype.slice.call(convoCards).forEach(convo => {
+
+    //for each card, add event listener to be triggered when user scrolls down a percentage of its height
+    window.addEventListener('scroll', function(){
```

Comment:
Instead of adding multiple scroll listeners, it would be more efficient to add one and then do your `.forEach(convo => {` inside.

Even better, for scroll performance, you should probably add a debounce so the calculations aren't happening continually.

Another option is rather than use scrolling, use `requestAnimationFrame`.

Or an even better option is to not do it yourself and use something like [inViewport](https://github.com/vvo/in-viewport).